index.js

```diff
@@ -21,30 +22,20 @@ const exec = (cmd, args) => {
 	).filter(Boolean);
 };
 
-const install = (bin, useYarn) => () => execa(bin, ['install'])
-	.catch(err => {
-		if (err.code === 'ENOENT' && useYarn) {
-			return execa('npm', ['install']);
-		}
-		throw err;
-	});
-
 module.exports = (input, opts) => {
 	input = input || 'patch';
 
 	opts = Object.assign({
 		cleanup: true,
-		publish: true
-	}, opts, {
-		yarn: opts && opts.yarn !== false
-	});
+		publish: true,
+		yarn: hasYarn()
+	}, opts);
```

Comment:
I think this would lead to confusing behavior. If the user wants to switch to Yarn, but has yet to generate a lockfile, it would seem as if this library isn't working.

I think the current implementation is fine, yet if you really want to ensure a lockfile is present, we should throw an error and alert the user what the issue is.